sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -166,6 +207,45 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     });
   }
 
+  /**
+   * Creates a ministry notification for the saved education
+   * program offering as a part of the same transaction.
+   * @param offeringName offering name.
+   * @param programName related program name.
+   * @param programOfferingStatus related program offering status.
+   * @param institutionLocationId related institution location id.
+   * @param institutionLocationName related institution location name.
+   * @param entityManager entity manager to be part of the transaction.
+   */
+  private async saveEducationProgramOfferingNotification(
+    offeringName: string,
+    programName: string,
+    programOfferingStatus: OfferingStatus,
+    institutionLocationId: number,
+    institutionLocationName: string,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    const institutionLocation =
+      await this.institutionLocationService.getInstitutionLocation(
```

Comment:
This method is called inside a `for` for the `bulk-insert`, right?
Is this not the call that we would like to avoid?
Can we not have the data needed loading ahead of time or is this not the one that we talked about before?